account_check_printing_report_base/models/account_payment.py

```diff
@@ -36,8 +36,10 @@ def do_print_checks(self):
 
     @api.multi
     def post(self):
-        res = super(AccountPayment, self).post()
-        if (self.journal_id.check_print_auto and
-                self.payment_method_id.code == 'check_printing'):
-            return self.do_print_checks()
+        for rec in self:
+            res = super(AccountPayment, rec).post()
```

Comment:
Don't nest `super` call

Reply:
@pedrobaeza , I have about 3 clients having issues due to this function not being properly implemented to handle multiple payments. 

"Error:
Odoo Server Error

Traceback (most recent call last):
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/odoo/odoo-server/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 936, in _call_
    return self.method(*args, **kw)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/odoo/odoo-server/src/odoo/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/odoo/odoo-server/src/odoo/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/addons/account/models/account_payment.py", line 235, in create_payments
    payments.post()
  File "/odoo/odoo-server/src/custom/iterativo/dominicana-premium/invoice_payment_method/models/account_payment.py", line 44, in post
    res = super(AccountPayment, self).post()
  File "/odoo/odoo-server/src/custom/oca/account-payment/account_check_printing_report_base/models/account_payment.py", line 40, in post
    if (self.journal_id.check_print_auto and
  File "/odoo/odoo-server/src/odoo/odoo/fields.py", line 935, in _get_
    record.ensure_one()
  File "/odoo/odoo-server/src/odoo/odoo/models.py", line 4366, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: account.payment(1613, 1614)"

Reply:
You only have to let the super call as it was, and after that, do the for loop

Reply:
The thing is, that it doesn't even get to the loop if i don't make the super call inside it. It raise an error at the moment of the assignment. when i debugged, here was where the process stopped. 
[   res = super(AccountPayment, self).post() ]

Reply:
But then the problem is not in this module, but in a super one.

Reply:
I made the changes and tested it and it worked. will update this. thanks .!!